rfc-163-cdn-rationalisation.md

```diff
@@ -0,0 +1,85 @@
+- A single customisation point is exposed to us: the `fetch` event, for which we can provide a handler in the language of our choice
+- It's our responsibility to construct a response for the end user, by making one or more requests to the backend(s) of our choice, or by responding with a hard-coded template, whichever is appropriate.
+
+![Sequence diagram showing a request being handled on Fastly Compute@Edge](rfc-163/Compute%40Edge%20sequence%20diagram.png)
+
+The below sequence diagram illustrates how backend failover works on Compute@Edge. Again, note that it's our own code that performs the failover.
+
+![Sequence diagram showing backend failover on Fastly Compute@Edge](rfc-163/Compute%40Edge%20backend%20failover.png)
+
+#### Known limitations
+
+Compute@Edge has a couple of known limitations that _may_ affect this work, depending on how much we care about these features:
+
+> Compute@Edge services currently offer a fetch API that performs a backend fetch through the Fastly edge cache, and stores cacheable responses. There is no way to adjust cache rules for objects received from a backend before they are inserted into the cache within a Compute@Edge service. As a result, if you need to process received objects before caching them, or to set custom cache TTLs, a solution is to place a VCL service in a chain with a Compute@Edge one
+>
+> -- <cite>https://developer.fastly.com/learning/concepts/service-chaining/#computeedge-to-vcl-chaining</cite>
+
+This means that it's currently not possible to implement all of the code from our `vcl_fetch` subroutine within a Compute@Edge service, including e.g. [this code](https://github.com/alphagov/govuk-cdn-config/blob/b0f104094c34c9b72c311a7aae2f04603364a1f1/vcl_templates/www.vcl.erb#L527-533) which marks a response as uncacheable if the `GOVUK-Account-Session` or `GOVUK-Account-End-Session` headers are set.
+
+This code was introduced in https://github.com/alphagov/govuk-cdn-config/pull/357, and is defence-in-depth to catch the case where appropriate caching headers were not set by origin (e.g. `Vary: GOVUK-Account-Session` or `Cache-Control: no-store`).
+
+Fastly have plans to address this limitation in the future, but in the meantime a workaround is to ["chain" a VCL service behind our Compute@Edge one](https://developer.fastly.com/learning/concepts/service-chaining/#computeedge-to-vcl-chaining). If we wish to retain this defence-in-depth functionality, this is how we'll need to implement it.
+
+---
+
+> Currently, Compute@Edge does not support shielding, making service chaining a useful mechanism for adding shielding to C@E services
+>
+> -- <cite>https://developer.fastly.com/learning/concepts/service-chaining/#chaining-with-shielding-in-computeedge</cite>
+
+We're currently using shielding on `gov.uk/alerts`, so we would need to make use of service chaining to allow this to continue working when we migrate the rest of our logic to Compute@Edge.
+
+### Standardise on Terraform for CDN configuration
```

Comment:
It looks like @samsimpson1 from Platform Engineering is already looking into this, but is currently on leave - I will check in when he returns and update this section accordingly